src/spi.rs

```diff
@@ -83,88 +85,99 @@ where
     }
 
     fn get_fw_version(&mut self) -> Result<FirmwareVersion, self::Error> {
-        self.control_pins.wait_for_esp_select();
-
-        self.send_cmd(NinaCommand::GetFwVersion, 0).ok().unwrap();
+        // TODO: improve the ergonomics around with_no_params()
+        let operation =
+            Operation::new(NinaCommand::GetFwVersion, 1).with_no_params(NinaNoParams::new(""));
 
-        self.control_pins.esp_deselect();
-        self.control_pins.wait_for_esp_select();
+        self.execute(&operation);
 
-        let bytes = self
-            .wait_response_cmd(NinaCommand::GetFwVersion, 1)
-            .ok()
-            .unwrap();
-        self.control_pins.esp_deselect();
+        let result = self.receive(&operation)?;
 
-        Ok(FirmwareVersion::new(bytes)) // e.g. 1.7.4
+        Ok(FirmwareVersion::new(result)) // e.g. 1.7.4
     }
 
     fn set_passphrase(&mut self, ssid: &str, passphrase: &str) -> Result<(), self::Error> {
-        self.control_pins.wait_for_esp_select();
+        let operation = Operation::new(NinaCommand::SetPassphrase, 1)
+            .param(NinaSmallArrayParam::new(ssid))
+            .param(NinaSmallArrayParam::new(passphrase));
+
+        self.execute(&operation);
+
+        self.receive(&operation);
```

Comment:
Can we use the result of `receive` as the result to return from this function?

Reply:
The plan I had in my head was to think more deeply about all of these returns as part of the work for https://github.com/Jim-Hodapp-Coaching/esp32-wroom-rp/issues/21 as it is also scheduled as part of this `v0.2` release.  Let me know if you'd prefer making the updates as part of this PR.

Reply:
This works for me, thanks for clarifying.